### PR TITLE
`/newtask` 

### PR DESCRIPTION
**Pull request by Keyrxng**
Resolves https://github.com/ubiquity-os-marketplace/ubiquity-os-kernel-telegram/issues/32

this PR introduces the `/newtask` command which is used by bot admins to create new tasks in the correct repository.

Usage: `/newtask ubiquity-os-marketplace/ubiquity-os-kernel-telegram`

- The admin must reply to the message which they want to turn into a task spec
- They must include the `repo.full_name` after the command in order to open it in the correct repo
- Only admins are able to use this new command
- New tasks are currently opened under my alt account because the org secret `APP_PRIVATE_KEY` issue, no one has replied to me about it so I'm assuming it's still unresolved. It needs fixed ASAP.

QA: https://github.com/ubq-testing/telegram--bot/issues/9

Task creation comment:

> The plugin DM interface needs to be standardized so that all plugins can easily use it to enable notifications for any reason.

> Not just that but the logger can have a plugin built to enable it to be able to DM errors too. That feels like covering both the good and bad outcomes regarding notifications

image

### CODE REVIEW

**Comment by Keyrxng** [comment]
1. Using GPT to structure a proper spec. 
2. Using `Directives` and `Constraints` to guide, please offer suggestions on improvements.
3. With embeddings I expect the specs to be more insightful and less generic.

Broke this into it's own PR and task via a task list, I think that's the right thing to do here. Once I get the embeddings data I can make moves on the other.

**Comment by Keyrxng** [comment]
I saw the thumbs up on the task to my comment so I'll take that as a go-ahead and I''ll merge this today

**Comment by 0x4007** [comment]
> Usage: /newtask ubiquity-os-marketplace/ubiquity-os-kernel-telegram

This is very hostile UX. Perhaps we should fuzzy match so we can write something like "/newtask kernel-telegram" and it'll pick from a list of the repos in the devpool-directory.

File: src/types/plugin-inputs.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ export const pluginSettingsSchema = T.Object({
     .Encode((value) => value.toString()),
   shouldUseGithubStorage: T.Boolean({ default: false }),
   storageOwner: T.String({ default: "ubiquity-os-marketplace" }),
+  maxCompletionTokens: T.Number({ default: 7000 }),
```
Comment (0x4007) [inline src/types/plugin-inputs.ts:23]: Why 7000

Comment (Keyrxng) [reply src/types/plugin-inputs.ts:23]: `/ask` defaults to 10k, it was 8k before and telegram with huge blocks of texts is uneasy on the eyes at times

https://github.com/ubiquity-os-marketplace/command-ask/blob/782c838eae61d977133d4b071987cb213ab6182a/src/types/plugin-inputs.ts#L26

**Comment by Keyrxng** [comment]
> > Usage: /newtask ubiquity-os-marketplace/ubiquity-os-kernel-telegram
> 
> This is very hostile UX. Perhaps we should fuzzy match so we can write something like "/newtask kernel-telegram" and it'll pick from a list of the repos in the devpool-directory.

Could do that no probs

I had planned on using the inline feature to list the repos and you just select one, it sends and we post on that message' update but fuzzy sounds good.

File: src/types/plugin-inputs.ts (hunk 1 above)
Comment (0x4007) [reply src/types/plugin-inputs.ts:23]: I think if you don't put anything it defaults to max which makes more sense to use.

Comment (Keyrxng) [reply src/types/plugin-inputs.ts:23]: cool will do.

Can you price this task for me?

And is there any chance you could message me on TG?

**Comment by Keyrxng** [comment]
> cool will do.
> 
> Can you price this task for me?
> 
> And is there any chance you could message me on TG?

I noticed you priced the task but decided to not message me on telegram, idk why you are ignoring me but there are things that need cleared up:

1. Previously I was merging my PRs into `development` myself and without proper review, Rndquu told me to stop that until you had replied to me and let me know the circumstances, although progress will continue to be very slow if that's the case.
2. The ORG readme shows `kernel-telegram: Keyrxng` implying I should be building this instead of `plugin-installer` but:
 
- You are not pricing tasks and I am not to work on anything unpriced
- Reviews are way too slow to gain any sort of momentum I imagine due to the complexity of the plugin/workload of reviewers

Can you let me know what's going on please as I've been left in the dark for months now and a little bit of communication would go a long way, thanks.

**Comment by Keyrxng** [comment]
For the sake of QA I didn't create the issue in the repo instead just logged where it would have been created in the reply in the footer:

- `/newtask pay.ubq`
image


- `/newtask pricing` >> `ubiquity-os-marketplace/daemon-pricing`
image

**Pull request merged**